### PR TITLE
sub headings not getting direct linkified

### LINKED ISSUE #2736: Sub headings not getting direct linkified
I noticed these two `<h2>` tags aren't being highlighted and turned into a direct link like they should be. This is happening on the [Using CSS Variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties) page.

Screen Shot 2021-01-28 at 8 13 53 PM 

Screen Shot 2021-01-28 at 8 15 57 PM

Screen Shot 2021-01-28 at 8 16 46 PM
Screen Shot 2021-01-28 at 8 13 35 PM

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #2736

Good pages to test:

* http://localhost:3000/en-US/docs/Web/CSS/Using_CSS_custom_properties#_flaws
* http://localhost:3000/en-US/docs/Learn/Accessibility/Mobile#_flaws
* http://localhost:3000/en-us/docs/learn/html/howto#_flaws

Or you can start `yarn dev` in one terminal, run `yarn build` in another and keep an eye on http://localhost:3000/en-US/_flaws?flaws=sectioning in the browser. You'll see something like this:
Screen Shot 2021-01-29 at 5 18 43 PM

I rushed to finish this on Friday and I haven't had time to write any end-to-end tests so I'll work on that on Monday.

### CODE REVIEW

**Comment by peterbe** [comment]
@escattone I made some unrelated changes to the build CLI so I can get quiet down the logging so I can see only the interesting messages on my terminal when I run `yarn build`. Sorry, but it's pretty harmless.

File: build/cli.js [hunk 1]
```diff
@@ -180,7 +180,7 @@ async function buildDocuments(files = null) {
 
     if (!options.noProgressbar) {
       progressBar.increment();
-    } else {
+    } else if (!quiet) {
```
Comment (Ryuno-Ki) [inline build/cli.js:183]: Perhaps worth to think about log levels and have an environment controlling the minimum level being logged …

**Comment by peterbe** [comment]
@escattone Can you help me look into http://localhost:3000/en-US/docs/Web/Embeddable#_flaws
Is that page sane? There you have multiple `<h2>`s that sit *not* at the root level. Somehow the page manages to extract the first `<h2>` from the `<div class="hidden">` to give the page a title.

File: build/document-extractor.js [hunk 2]
```diff
@@ -322,22 +333,50 @@ function _addSectionProse($) {
...
   const h2s = $.find("h2");
-  if (h2s.length === 1) {
-    id = h2s.attr("id");
-    title = h2s.html();
-    titleAsText = h2s.text();
-    h2s.remove();
-  } else {
+  for (const i of [...Array(h2s.length).keys()]) {
+    if (i) {
+      // Excess!
```
Comment (Ryuno-Ki) [inline build/document-extractor.js:345]: As in „more than one“?

File: build/document-extractor.js [hunk 3]
```diff
@@ -322,22 +333,50 @@ function _addSectionProse($) {
...
+  if (!h2found) {
     const h3s = $.find("h3");
-    if (h3s.length === 1) {
-      id = h3s.attr("id");
-      title = h3s.html();
-      titleAsText = h3s.text();
-      if (id && title) {
-        isH3 = true;
-        h3s.remove();
+    for (const i of [...Array(h3s.length).keys()]) {
```
Comment (Ryuno-Ki) [inline build/document-extractor.js:364]: That feels like being ripe for refactoring down the road.
The logic is quite similar to the one used for `h2`

**Review by escattone — APPROVED** [review]

**Comment by peterbe** [comment]
I was able to use this new code to find that [these pages](https://github.com/mdn/content/pull/2035) needed to fixed. The big flashlight!

**Pull request merged**